content/post/altswiftui-travel/index.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "Introducing AltSwiftUI in Rakuten Travel"
+date: 2021-07-06T18:18:13+09:00
+excerpt: "What is AltSwiftUI and how we introduced it in Rakuten Travel"
+timeToRead: 4
+authors:
+   - Kevin Wong
+hero: /post/altswiftui-travel/hero.png
+draft: true
+license: cc-by
+---
+
+_Background image by [Dan Roizer](https://unsplash.com/@danroizer) under the [Unsplash license](https://unsplash.com/license)_
+
+SwiftUI, introduced by Apple in 2019, allows developers to create apps with declarative UI programming and state management, which greatly helps to improve development speed and quality. However, the framework is locked at the iOS 13 SDK, and every new improvement iteration to this new framework will push up this restriction to the latest SDK (iOS 14, and so on).
+
+AltSwiftUI is an open source UI framework for iOS that mimics SwiftUI’s interface, introducing a single OS baseline at iOS 11 and greater flexibility because of its open source nature.
+
+Starting to use AltSwiftUI is pretty straightforward, just import it through Cocoa pods or Swift Package Manager. If you are familiar with SwiftUI syntax, then you are good to go.
+
+For more info see: https://github.com/rakutentech/AltSwiftUI
+
+## A peak into code
+
+Other than some small differences, the code structure is highly similar to that of regular SwiftUI code. Of the main things we can point out, AltSwiftUI doesn't use associated types in views, meaning that it is not necessary to use opaque return types (`some View`).
+
+![AltSwiftUI code sample](altswiftui-code-sample.png)
+
+## Extended features
+
+AltSwiftUI interface is inspired by SwiftUI, but we needed more features and capabilities than originally offered, so we introduced extra functionalities in AltSwiftUI.
+
+For example, we needed deeper integrations with List and ScrollView interactions, so there are methods like `ScrollView.scrollOffset` and `List.onDragStarted`. There are also multiple small additions to some of the views and modifiers, and this list will keep growing in the future. To identify these features, you can read the source documentation during development or at https://altswiftui.com.
+
+
+## AltSwiftUI in Rakuten Travel
+
+AltSwiftUI’s development started at the end of 2019, with the main purpose of taking on new projects while renewing the way we design software, improve the speed and quality of our development, and be more futureproof. During this journey, we tackled multiple challenges by comparing what SwiftUI originally offered with what our app needed, and the open source nature of this framework allowed us to improve it by adding features that would allow us to meet our app expectations. 
+
+_Rakuten Travel Premium Club_ is an invitation only app built with AltSwiftUI.
+
+### Simplified Architecture
```

Comment:
Is the MV model a feature of SwiftUI or AltSwiftUI? I'm not an iOS developer so it might be clear to iOS developers but it isn't very clear for me.

Reply:
MV is a derivation of the most commonly known MVC design pattern used by most iOS developers. SwiftUI and consequently AltSwiftUI help achieving this kind of pattern.